includes/wc-template-functions.php

```diff
@@ -241,7 +268,7 @@ function wc_set_loop_prop( $prop, $value = '' ) {
  * @return bool
  */
 function woocommerce_product_loop() {
-	return have_posts() || 'products' !== woocommerce_get_loop_display_mode();
+	return wc_get_loop_prop( 'total' ) > 0 || 'products' !== woocommerce_get_loop_display_mode();
```

Comment:
`wc_get_loop_prop()` it set again the loop, also `have_posts()` works different, and probably better keep it for backwards compatibility.

Reply:
I don't remember why I made this change, maybe I needed it while I was in the middle of the development; but I have verified that everything still works after reverting to using `have_posts` again so I did and pushed that.